supply/lib/supply/uploader.rb

```diff
@@ -7,46 +7,115 @@ def perform_upload
 
       verify_config!
 
-      if metadata_path
-        UI.user_error!("Could not find folder #{metadata_path}") unless File.directory?(metadata_path)
-
-        all_languages.each do |language|
-          next if language.start_with?('.') # e.g. . or .. or hidden folders
-          UI.message("Preparing to upload for language '#{language}'...")
-
-          listing = client.listing_for_language(language)
-
-          upload_metadata(language, listing) unless Supply.config[:skip_upload_metadata]
-          upload_images(language) unless Supply.config[:skip_upload_images]
-          upload_screenshots(language) unless Supply.config[:skip_upload_screenshots]
-          upload_changelogs(language) unless Supply.config[:skip_upload_metadata]
-        end
-      end
-
       apk_version_codes = []
       apk_version_codes.concat(upload_apks) unless Supply.config[:skip_upload_apk]
       apk_version_codes.concat(upload_bundles) unless Supply.config[:skip_upload_aab]
       upload_mapping(apk_version_codes)
 
       apk_version_codes.concat(Supply.config[:version_codes_to_retain]) if Supply.config[:version_codes_to_retain]
 
+
       # Only update tracks if we have version codes
       # Updating a track with empty version codes can completely clear out a track
       update_track(apk_version_codes) unless apk_version_codes.empty?
 
+
+      if !Supply.config[:rollout].nil? && Supply.config[:version_code].to_s != "" && Supply.config[:track].to_s != ""
```

Comment:
Clean this logic up - hard to follow

Reply:
Is `version_code` needed? Maybe error out in `update_rollout` if multiple versions and `:version_code` is needed